microsoft/testsuites/kdump/kdumpcrash.py

```diff
@@ -390,76 +471,38 @@ def _check_kdump_result(
-                    if incomplete_file:
-                        check_dump_file_tries = 0
-                        stat = node.tools[Stat]
-                        incomplete_file_size = stat.get_total_size(incomplete_file)
-                except Exception as identifier:
-                    log.debug(
-                        "Fail to execute command. It may be caused by the system kernel"
-                        " reboot after dumping vmcore."
-                        f"{identifier.__class__.__name__}: {identifier}. Retry..."
-                    )
-                    # Hit exception, break this loop and re-try to connect the system
-                    break
-                if incomplete_file:
-                    # If the incomplete file doesn't grow in 100s, then raise exception
-                    if incomplete_file_size > saved_dumpfile_size:
-                        saved_dumpfile_size = incomplete_file_size
-                        check_incomplete_file_tries = 0
-                    else:
-                        check_incomplete_file_tries += 1
-                        if check_incomplete_file_tries >= max_tries:
-                            serial_console.get_console_log(
-                                saved_path=log_path, force_run=True
-                            )
-                            node.execute("df -h")
-                            raise LisaException(
-                                "The vmcore file is incomplete with file size"
-                                f" {round(incomplete_file_size/1024/1024, 2)}MB"
-                            )
-                else:
-                    # If there is no any dump file in 100s, then raise exception
-                    check_dump_file_tries += 1
-                    if check_dump_file_tries >= max_tries:
-                        serial_console.get_console_log(
-                            saved_path=log_path, force_run=True
-                        )
-                        raise LisaException(
-                            "No vmcore or vmcore-incomplete is found under "
-                            f"{kdump.dump_path} with file size greater than 10M."
-                        )
-                if timer.elapsed(False) > self.timeout_of_dump_crash:
-                    serial_console.get_console_log(saved_path=log_path, force_run=True)
-                    raise LisaException("Timeout to dump vmcore file.")
-                time.sleep(5)
-        serial_console.get_console_log(saved_path=log_path, force_run=True)
-        raise LisaException("Timeout to connect the VM after triggering kdump.")
+            dump_file_found = self._wait_dump_file_generated(
+                node=node,
+                log_path=log_path,
+                log=log,
+                kdump=kdump,
+                timer=timer,
+                serial_console=serial_console,
+            )
+            if dump_file_found:
+                return
+        self._get_console_log(
```

Comment:
How the ready platform supported? It looks still raising exceptions for ready platform. How about write a mock SerialConsole support for ready platform? It do nothing for save serial log, but if some test cases need to ready the console log, raise an exception.

Reply:
updated